source/preprocessor.tex

```diff
@@ -671,6 +676,9 @@
 \indextext{header unit!preprocessing}%
 \indextext{macro!import|(}%
 
+\pnum
+An \grammarterm{import-line} shall be of the form
```

Comment:
For "of the form" constraints, we usually don't show another non-terminal, but just the grammar sequence that needs to be satisfied. That said, I think the overall change is a net improvement.

Reply:
I noticed that, but the subsequent text refers to things like "all three forms of pp-import" etc.  So I thought it nice to keep that name.  I suppose we could change it to "all three forms above" and "the second form", etc.  Not sure if that would be better though.

Reply:
We do the same thing in [class.conv.fct], which is much more egregious than this (it lists three productions and only means the first one). Though I do wonder if we can do better here; perhaps we could list *pp-import* as a *control-line* production (perhaps renaming it to *import-line*), and change [cpp.pre]/2 to say that a *text-line* is only matched if the line is not a preprocessor directive. (That way, an import directive that's not a *pp-import* would simply not match the preprocessor grammar, just like an unmatched `#if` or `#endif`.)